presto-main/src/main/java/com/facebook/presto/operator/Operator.java

```diff
@@ -52,7 +52,7 @@
      */
     default ListenableFuture<?> isBlocked()
     {
-        return NOT_BLOCKED;
+        return getOperatorContext().isWaitingForMemory();
```

Comment:
Instead of modifying every operator to deal with `operatorContext.isWaitingForMemory()`, why not have `Driver` call `operator.getOperatorContext().isWaitingForMemory()` when checking `isBlocked`?